packages/geometry/src/rectangle.ts

```diff
@@ -87,6 +87,23 @@ export const cropFit = (to: Rectangle, rect: Rectangle): Rectangle => {
   return fromPointAndDimensions(position, dimensions);
 };
 
+/**
+ * Returns a rectangle where each side of `rect` will be reduced proportionally
+ * to have an area less than or equal to the provided `to` value. The returned
+ * rectangle will be centered within the original bounds of `rect`.
+ *
+ * @param to - the maximum area this rectangle can have
+ * @param rect - the rectangle to scale to fit the specified area
+ */
+export const scaleFit = (to: number, rect: Rectangle): Rectangle => {
+  const scale = Math.min(Math.sqrt(to / area(rect)), 1);
+  const dimensions = Dimensions.floor(
+    Dimensions.proportionalScale(scale, rect)
+  );
+  const position = Point.subtract(center(rect), Dimensions.center(dimensions));
```

Comment:
is there potential for alternate behaviors for the resulting position after the `scaleFit`? like keeping one of the same points? or is the scaling down and moving the behavior we always want for now?

Reply:
Good question. As far as I know right now the re-centering behavior is always what we want, but that's just based on the existing behavior also re-centering, so maybe it's worth revisiting that

Reply:
sounds like something we can revisit if we need different behavior.